src/modes/coincident_select.js

```diff
@@ -83,13 +83,26 @@ CoincidentSelect.onSetup = function(opts) {
   const planId = features.find(f => f.properties.vetro_id === feature.id)
     .properties.plan_id;
 
-  features.forEach(f => {
+  for(const f of features){
     if (
       f.properties.plan_id === planId &&
       f.geometry.type === "LineString" &&
       !f.layer.id.includes("_snap")
     ) {
-      const lineGeom = JSON.parse(f.properties.geojson_string);
+
+      let lineGeom = f.geometry;
+      if(this._ctx.options.fetchSourceGeometry && typeof this._ctx.options.fetchSourceGeometry === "function"){
```

Comment:
minor suggestion: I think `typeof this._ctx.options.fetchSourceGeometry === "function"` works fine without the check for undefined first